Invalid value for URL param `order` should cause 4xx error

Currently an invalid value given to the `order` URL param will produce a 500 response along with a stack trace.
Instead, a 4xx response should be returned with no stack trace.

Example:
```
$ curl --fail-with-body 'https://service.dev.singlecell.gi.ucsc.edu/index/bundles?catalog=dcp2&filters=%7B%7D&order=%2527&search_after=%5B1644398658267%2C%20%22494295a9-d6da-42a6-bda2-40b51ffeb84d%22%5D&size=10&sort=bundleVersion'
Traceback (most recent call last):
…
  File "/var/task/azul/service/elasticsearch_service.py", line 132, in prepare_request
    request = self.outer.prepare_request(request)
  File "/var/task/azul/service/elasticsearch_service.py", line 515, in prepare_request
    request = request.sort(*sort(sort_order))
  File "/var/task/azul/service/elasticsearch_service.py", line 485, in sort
    assert order in ('asc', 'desc'), order
AssertionError: %27
curl: (22) The requested URL returned error: 500
```

- [ ] Security design review completed; the Resolution of this issue does **not** …
  - [ ] … affect authentication; for example:
    - OAuth 2.0 with the application (API or Swagger UI)
    - Authentication of developers with Google Cloud APIs
    - Authentication of developers with AWS APIs
    - Authentication with a GitLab instance in the system
    - Password and 2FA authentication with GitHub
    - API access token authentication with GitHub
    - Authentication with 
  - [ ] … affect the permissions of internal users like access to
    - Cloud resources on AWS and GCP
    - GitLab repositories, projects and groups, administration
    - an EC2 instance via SSH
    - GitHub issues, pull requests, commits, commit statuses, wikis, repositories, organizations
  - [ ] … affect the permissions of external users like access to
    - TDR snapshots
  - [ ] … affect permissions of service or bot accounts
    - Cloud resources on AWS and GCP
  - [ ] … affect audit logging in the system, like
    - adding, removing or changing a log message that represents an auditable event
    - changing the routing of log messages through the system
  - [ ] … affect monitoring of the system
  - [ ] … introduce a new software dependency like
    - Python packages on PYPI
    - Command-line utilities
    - Docker images
    - Terraform providers
  - [ ] … add an interface that exposes sensitive or confidential data at the security boundary
  - [ ] … affect the encryption of data at rest
  - [ ] … require persistence of sensitive or confidential data that might require encryption at rest
  - [ ] … require unencrypted transmission of data within the security boundary
  - [ ] … affect the network security layer; for example by 
    - modifying, adding or removing firewall rules
    - modifying, adding or removing security groups
    - changing or adding a port a service, proxy or load balancer listens on
- [ ] Documentation on any unchecked boxes is provided in comments below

--- a/lambdas/service/app.py
+++ b/lambdas/service/app.py
@@ -672,7 +672,7 @@ def validate_repository_search(params, **validators):
     validate_params(params, **{
         'catalog': validate_catalog,
         'filters': validate_filters,
-        'order': str,
+        'order': validate_order,
         'search_after': str,
         'search_after_uid': str,
         'search_before': str,
@@ -795,6 +795,12 @@ def validate_manifest_format(format_: str):
         if format_ not in supported_formats:
             raise BRE(f'Manifest format `{format_}` is not supported for '
                       f'catalog {app.catalog}. Must be one of {supported_formats}')
+
+
+def validate_order(order: str):
+    supported_orders = ('asc', 'desc')
+    if order not in supported_orders:
+        raise BRE(f'Unknown order `{order}`. Must be one of {supported_orders}')
 
 
 class Mandatory:

--- a/test/service/test_request_validation.py
+++ b/test/service/test_request_validation.py
@@ -32,7 +32,20 @@ def setUpModule():
     configure_test_logging()
 
 
+@patch_dss_source
+@patch_source_cache
 class RequestParameterValidationTest(WebServiceTestCase):
+    maxDiff = None
+
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls._setup_indices()
+
+    @classmethod
+    def tearDownClass(cls):
+        cls._teardown_indices()
+        super().tearDownClass()
 
     def assertResponseStatus(self, url: furl, status: int) -> Response:
         response = requests.get(str(url))
@@ -328,3 +341,17 @@ class RequestParameterValidationTest(WebServiceTestCase):
             with self.subTest(size=size):
                 url.args.set('size', size)
                 test(url, arg)
+
+    def test_order(self):
+        url = self.base_url.set(path='/index/projects')
+        for order, arg in [
+            ('foo', "Unknown order `foo`. Must be one of ('asc', 'desc')"),
+            ('asc', None),
+            ('desc', None)
+        ]:
+            with self.subTest(order=order):
+                url.args.set('order', order)
+                if arg:
+                    self.assertBadRequest(url, arg)
+                else:
+                    self.assertResponseStatus(url, 200)

